mlflow/models/utils.py

```diff
@@ -97,18 +93,27 @@ def _is_scalar(x):
                 "(pandas.DataFrame, numpy.ndarray, dict, list), "
                 "got {}".format(type(input_example))
             )
+
         example_filename = "input_example.json"
-        self.data = input_example.to_dict(orient="split")
-        # Do not include row index
-        del self.data["index"]
-        if all(input_example.columns == range(len(input_example.columns))):
-            # No need to write default column index out
-            del self.data["columns"]
-        self.info = {
-            "artifact_path": example_filename,
-            "type": "dataframe",
-            "pandas_orient": "split",
-        }
+        if isinstance(input_example, pd.DataFrame):
+            self.data = input_example.to_dict(orient="split")
+            # Do not include row index
+            del self.data["index"]
+            if all(input_example.columns == range(len(input_example.columns))):
+                # No need to write default column index out
+                del self.data["columns"]
+            self.info = {
+                "artifact_path": example_filename,
+                "type": "dataframe",
+                "pandas_orient": "split",
+            }
+        else:
+            self.data = {"instances": input_example}
```

Comment:
you probably want to call input_example.tolist().

We also want to handle dictionary of tensors, the easies would be to use the `instances`  columnar representation, so maybe lets use instances for both numpy array and dict of numpy arrays?